release-26.2.1-rc: kv/kvserver: TestStoreMetrics failed

kv/kvserver.TestStoreMetrics [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21354325?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21354325?buildTab=artifacts#/) on release-26.2.1-rc @ [7797103d26398f23c60599f86df401d6454dadb0](https://github.com/cockroachdb/cockroach/commits/7797103d26398f23c60599f86df401d6454dadb0):


```
=== RUN   TestStoreMetrics
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/3fe72dd26006841c1f679405fff92eae/logTestStoreMetrics3500105969
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:146: using test tenant configuration from testserver factory defaults
    client_metrics_test.go:331: condition failed to evaluate within 45s: replica still exists on dest 0
        
        goroutine dump: /artifacts/tmp/_tmp/3fe72dd26006841c1f679405fff92eae/goroutine_dump_2501131889.txt
    panic.go:615: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/3fe72dd26006841c1f679405fff92eae/logTestStoreMetrics3500105969
--- FAIL: TestStoreMetrics (53.91s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStoreMetrics.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-64112

--- a/pkg/kv/kvserver/client_metrics_test.go
+++ b/pkg/kv/kvserver/client_metrics_test.go
@@ -327,7 +327,10 @@ func TestStoreMetrics(t *testing.T) {
 	verifyStatsOnServers(t, tc, specs, stickyVFSRegistry, 1, 2)
 	checkGauge(t, "store 0", tc.GetFirstStoreFromServer(t, 0).Metrics().ReplicaCount, initialCount+1)
 	tc.RemoveLeaseHolderOrFatal(t, desc, tc.Target(0), tc.Target(1))
+	// The removed replica may never learn about its own removal through raft,
+	// so force the replica GC queue to clean it up.
 	testutils.SucceedsSoon(t, func() error {
+		tc.GetFirstStoreFromServer(t, 0).MustForceReplicaGCScanAndProcess()
 		_, err := tc.GetFirstStoreFromServer(t, 0).GetReplica(desc.RangeID)
 		if err == nil {
 			return fmt.Errorf("replica still exists on dest 0")

